tests/subsys/power/power_mgmt/src/main.c

```diff
@@ -0,0 +1,216 @@
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_false(device_power_state == DEVICE_PM_ACTIVE_STATE, NULL);
+#endif
+	zassert_true(set_sys_power == true, "App doesn't be notified");
+	zassert_false(state == SYS_POWER_STATE_ACTIVE, "Can't suspend system");
+}
+
+__weak void _sys_pm_power_state_exit_post_ops(enum power_states state)
+{
+	/* _sys_suspend is entered with irq locked
+	 * unlock irq before leave _sys_suspend
+	 */
+	irq_unlock(0);
+}
+
+__weak bool sys_pm_policy_low_power_devices(enum power_states pm_state)
+{
+	return sys_pm_is_sleep_state(pm_state);
+}
+
+/* Our PM policy handler */
+enum power_states sys_pm_policy_next_state(int ticks)
+{
+	enum power_states state;
+
+	zassert_true(z_is_idle_thread_object(_current), NULL);
+	zassert_true(ticks == _kernel.idle, NULL);
+
+	if (enter_low_power) {
+		enter_low_power = false;
+		notify_app_entry = true;
+		state = SYS_POWER_STATE_SLEEP_1;
+	} else {
+		/* only test sys_pm_policy_next_state()
+		 * no PM operation done
+		 */
+		state = SYS_POWER_STATE_ACTIVE;
+	}
+	return state;
+}
+
+/* implement in application, called by idle thread */
+void sys_pm_notify_power_state_entry(enum power_states state)
+{
+	TC_PRINT("enter suspend\n");
+	zassert_true(notify_app_entry == true, "App doesn't be notified");
+	zassert_true(z_is_idle_thread_object(_current), NULL);
+	zassert_equal(state, SYS_POWER_STATE_SLEEP_1, NULL);
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	struct device *dev;
+	uint32_t device_power_state;
+	/* at this point, devices are active */
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_equal(device_power_state, DEVICE_PM_ACTIVE_STATE, NULL);
+#endif
+	set_sys_power = true;
+	notify_app_exit = true;
+}
+
+/* implement in application, called by idle thread */
+void sys_pm_notify_power_state_exit(enum power_states state)
+{
+	TC_PRINT("leave suspend\n");
+	zassert_true(notify_app_exit == true, "App doesn't be notified");
+	zassert_true(z_is_idle_thread_object(_current), NULL);
+	zassert_equal(state, SYS_POWER_STATE_SLEEP_1, NULL);
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	struct device *dev;
+	uint32_t device_power_state;
+	/* at this point, devices are active again*/
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_equal(device_power_state, DEVICE_PM_ACTIVE_STATE, NULL);
+#endif
+	leave_idle = true;
+
+}
+
+/*
+ * @brief test power idle
+ *
+ * @details
+ *  - The global idle routine executes when no other work is available
+ *  - Provide a timeout parameter to the suspend routine indicating the
+ *    amount of time guaranteed to expire before the next timeout,
+ *    sys_pm_policy_next_state() handle this parameter.
+ *
+ * @see sys_pm_policy_next_state()
+ *
+ * @ingroup power_tests
+ */
+void test_power_idle(void)
+{
+	TC_PRINT("give way to idle thread\n");
+	k_sleep(SLEEP_TIMEOUT);
+}
```

Comment:
how we get know this case success or not?

Reply:
Please consider this throughout test cases. I anticipated seeing `return TC_PASS` or `return TC_FAIL` to indicate what @wentongwu raised here

Reply:
Power state change in idle thread, this case just give way to idle thread by k_sleep(). Success or not depends on
functions must be called in the idle thread such as sys_pm_policy_next_state(), flag "idle_entered" indicate that idle thread have entered and power management have happened.